tests/ui/filetype_is_file.rs

```diff
@@ -0,0 +1,23 @@
+#![warn(clippy::filetype_is_file)]
+
+fn main() -> std::io::Result<()> {
+    use std::ops::BitOr;
+    use std::fs;
+
+    // !filetype.is_dir()
+    if fs::metadata("foo.txt")?.file_type().is_file() {
```

Comment:
@flip1995 I have add two new case to be implemented, but didn't touch the origin lint logic, but why doesn't this line trigger lints?

Reply:
I don't really know why, whis should trigger the lint. Can you post the output of `dbg!(&args[0])` and `dbg!(ty)`?